libs/remix-core-plugin/src/lib/compiler-artefacts.ts

```diff
@@ -1,105 +1,83 @@
 'use strict'
-import { Plugin } from '@remixproject/engine'
-import { util } from '@remix-project/remix-lib'
-import { CompilerAbstract } from '@remix-project/remix-solidity'
+import {Plugin} from '@remixproject/engine'
+import {util} from '@remix-project/remix-lib'
+import {CompilerAbstract} from '@remix-project/remix-solidity'
 
 const profile = {
   name: 'compilerArtefacts',
-  methods: ['get', 'addResolvedContract', 'getCompilerAbstract', 'getAllContractDatas', 'getLastCompilationResult', 'getArtefactsByContractName', 'getContractDataFromAddress', 'getContractDataFromByteCode', 'saveCompilerAbstract'],
+  methods: ['get', 'addResolvedContract', 'getCompilerAbstract', 'getAllContractDatas', 'getLastCompilationResult', 'getArtefactsByContractName', 'getContractDataFromAddress', 'getContractDataFromByteCode', 'saveCompilerAbstract', 'getAllCompilerAbstracts'],
   events: [],
-  version: '0.0.1'
+  version: '0.0.1',
 }
 
 export class CompilerArtefacts extends Plugin {
   compilersArtefactsPerFile: any
   compilersArtefacts: any
-  constructor () {
+  constructor() {
     super(profile)
     this.compilersArtefacts = {}
     this.compilersArtefactsPerFile = {}
   }
 
-  clear () {
+  clear() {
     this.compilersArtefacts = {}
     this.compilersArtefactsPerFile = {}
   }
 
-  saveCompilerAbstract (file: string, compilerAbstract: CompilerAbstract) {
+  saveCompilerAbstract(file: string, compilerAbstract: CompilerAbstract) {
     this.compilersArtefactsPerFile[file] = compilerAbstract
   }
 
-  onActivation () {
-    const saveCompilationPerFileResult = (file, source, languageVersion, data, input?) => {
+  getAllCompilerAbstracts() {
+    return this.compilersArtefactsPerFile
+  }
+
+  onActivation() {
+    const saveCompilationResult = (file, source, languageVersion, data, input?) => {
       this.compilersArtefactsPerFile[file] = new CompilerAbstract(languageVersion, data, source, input)
+      this.compilersArtefacts.__last = this.compilersArtefactsPerFile[file]
     }
 
-    this.on('solidity', 'compilationFinished', (file, source, languageVersion, data, input, version) => {
-      this.compilersArtefacts.__last = new CompilerAbstract(languageVersion, data, source, input)
-      saveCompilationPerFileResult(file, source, languageVersion, data)
-    })
+    this.on('solidity', 'compilationFinished', saveCompilationResult)
```

Comment:
I think `saveCompilationResult` can take at most 5 parameters

Reply:
Yes it has 5 parameters?

https://github.com/sourcifyeth/remix-project/blob/fcc027579a01320244d761f7d95dae4a981cfc8e/libs/remix-core-plugin/src/lib/compiler-artefacts.ts#L36-L39